extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js

```diff
@@ -96,8 +97,44 @@ export class AmpStoryShoppingTag extends AMP.BaseElement {
     return layout === Layout_Enum.CONTAINER;
   }
 
+  /**
+   * This function counts the number of lines in the shopping tag
+   * and sets the styling properties dynamically based on the number of lines.
+   * @private
+   */
+  styleTagText_() {
+    const pillEl = this.element.shadowRoot?.querySelector(
+      '.amp-story-shopping-tag-pill'
+    );
+
+    const textEl = this.element.shadowRoot?.querySelector(
+      '.amp-story-shopping-tag-pill-text'
+    );
+
+    if (!pillEl || !textEl) {
+      return;
+    }
+
+    const fontSize = parseInt(
+      computedStyle(window, textEl).getPropertyValue('font-size'),
+      10
+    );
+    const ratioOfLineHeightToFontSize = 1.5;
+    const lineHeight = Math.floor(fontSize * ratioOfLineHeightToFontSize);
+    const height = textEl./*OK*/ clientHeight;
```

Comment:
Re: layout thrashing

this has to be in a `measureElement`
The order in which we read/write between the JS and the DOM really is a key component of web performance, it is important for us to get this mostly right in the codebase.

Reply:
I'll open up a new PR and address this issue.